hydromt/models/model_grid.py

```diff
@@ -374,8 +375,12 @@ def read_grid(self, fn: str = "grid/grid.nc", **kwargs) -> None:
             Additional keyword arguments to be passed to the `_read_nc` method.
         """
         self._assert_read_mode
-        for ds in self._read_nc(fn, **kwargs).values():
-            self.set_grid(ds)
+        # Load grid data in r+ mode to allow overwritting netcdf files
+        if self._read and self._write:
+            kwargs["load"] = True
+        for ds in self._read_nc(fn, single_var_as_array=False, **kwargs).values():
```

Comment:
I'm not 100% sure how python handle's function calls like this, I think it might be more efficient to do something like this: 
```suggesion
loaded_nc_files = self._read_nc(fn, single_var_as_array=False, **kwargs)
for ds in loaded_nc_filies.values():
```

Reply:
I dont know for sure but if you think it's better, we can change.